src/types.h

```diff
@@ -187,12 +187,17 @@ constexpr Value PieceValue[PIECE_NB] = {
 using Depth = int;
 
 enum : int {
-    DEPTH_QS_CHECKS    = 0,
-    DEPTH_QS_NO_CHECKS = -1,
-
-    DEPTH_NONE = -6,
-
-    DEPTH_OFFSET = -7  // value used only for TT entry occupancy check
+  // In QS, with depth <= 0, we search only "tactical" moves to fight the
```

Comment:
Fighting against the horizon effect is only a subproduct of qsearch, the idea is to have a more stable static evaluation of the position such that there are no tactical solutions that could render the static evaluation useless.

Reply:
i.e. if our evaluation function is as good in evaluating the tactical positions as it is good for quiet positions, we don't need no qsearch at all. you can just return eval.

Reply:
> the idea is to have a more stable static evaluation of the position such that there are no tactical solutions left, that could render the static evaluation useless.

this is part of what is meant by "the horizon effect" no?

Reply:
> we search only "tactical" moves to fight the horizon effect.

well, my nitpick is that searching (only "tactical" moves) does not fight against the horizon effect and hints that if you try any other moves you will get horizoned, which is not accurate, you will get an explosion and not neccessarly a horizon effect.

rather I find something like "because the horizon effect is more detrimental in the tactical positions we limit searching to them, otherwise we trust the evaluation function"
more to the point and less misleading.